sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java

```diff
@@ -1019,104 +1006,108 @@ public void startCopyFilePermissionFormat(FilePermissionFormat filePermissionFor
+        }).verifyComplete();
     }
 
     @Test
     public void startCopyWithOptionsLease() {
-        primaryFileAsyncClient.create(1024).block();
-        String sourceURL = primaryFileAsyncClient.getFileUrl();
-        String leaseId = createLeaseClient(primaryFileAsyncClient).acquireLease().block();
-        ShareRequestConditions conditions = new ShareRequestConditions().setLeaseId(leaseId);
+        Mono<ShareFileInfo> createFileMono = primaryFileAsyncClient.create(1024);
+        Mono<String> leaseIdMono = createFileMono.then(createLeaseClient(primaryFileAsyncClient).acquireLease());
 
-        ShareFileCopyOptions options = new ShareFileCopyOptions().setDestinationRequestConditions(conditions);
+        String sourceURL = primaryFileAsyncClient.getFileUrl();
 
-        PollerFlux<ShareFileCopyInfo, Void> poller
-            = setPlaybackPollerFluxPollInterval(primaryFileAsyncClient.beginCopy(sourceURL, options, null));
-        StepVerifier.create(poller)
-            .assertNext(it -> assertNotNull(it.getValue().getCopyId()))
-            .expectComplete()
-            .verify(Duration.ofMinutes(1));
+        StepVerifier.create(leaseIdMono.flatMapMany(leaseId -> {
+            ShareRequestConditions conditions = new ShareRequestConditions().setLeaseId(leaseId);
+            ShareFileCopyOptions options = new ShareFileCopyOptions().setDestinationRequestConditions(conditions);
+            PollerFlux<ShareFileCopyInfo, Void> poller
+                = setPlaybackPollerFluxPollInterval(primaryFileAsyncClient.beginCopy(sourceURL, options, null));
+            return poller.take(1).flatMap(response -> {
+                assertNotNull(response.getValue().getCopyId());
+                return Mono.just(response);
+            });
+        })).expectNextCount(1).verifyComplete();
     }
 
     @Test
     public void startCopyWithOptionsInvalidLease() {
-        primaryFileAsyncClient.create(1024).block();
+        Mono<ShareFileInfo> createFileMono = primaryFileAsyncClient.create(1024);
         String sourceURL = primaryFileAsyncClient.getFileUrl();
         String leaseId = testResourceNamer.randomUuid();
         ShareRequestConditions conditions = new ShareRequestConditions().setLeaseId(leaseId);
 
         ShareFileCopyOptions options = new ShareFileCopyOptions().setDestinationRequestConditions(conditions);
 
-        // exception: LeaseNotPresentWithFileOperation
-        assertThrows(ShareStorageException.class,
-            () -> setPlaybackPollerFluxPollInterval(primaryFileAsyncClient.beginCopy(sourceURL, options, null))
-                .blockFirst());
+        StepVerifier
+            .create(createFileMono.then(Mono.defer(
+                () -> setPlaybackPollerFluxPollInterval(primaryFileAsyncClient.beginCopy(sourceURL, options, null))
+                    .then(Mono.error(new ShareStorageException("LeaseNotPresentWithFileOperation", null, null))))))
+            .verifyError(ShareStorageException.class);
```

Comment:
This is not checking if the copy actually throws the error, as the error is thrown manually on line 1097. We want to change this to: 

```
StepVerifier
            .create(createFileMono.thenMany(
                setPlaybackPollerFluxPollInterval(primaryFileAsyncClient.beginCopy(sourceURL, options, null))))
            .verifyErrorSatisfies(r -> {
                ShareStorageException e = assertInstanceOf(ShareStorageException.class, r);
                assertEquals("LeaseNotPresentWithFileOperation", e.getErrorCode().getValue());
            });
```

This ensures that the copy actually throws the error.